tests/Calculator/LocaleAwareBcMathCalculatorTest.php

```diff
@@ -13,6 +15,6 @@ public function setUp(): void
     {
         parent::setUp();
 
-        $this->setLocale(LC_ALL, 'ru_RU.UTF-8');
+        setlocale(LC_ALL, 'ru_RU.UTF-8');
```

Comment:
I would add a file called `tests/LocaleTest.php`, with the following content.

```php
declare(strict_types=1);

namespace Tests\Money;

trait LocaleTest
{
    public static function setLocale(int $category, string $locale, \Closure $callback)
    {
        $currentLocale= \setlocale($category, 0);
        try {
            \setlocale($category, $locale);
            $callback();
        } finally {
            \setlocale($category, $currentLocale);
        }
    }
}
```

And then remove `setlocale` from `setUp` but simply use it in the test file as follows

```php
use Tests\Money\LocaleTest;

public static function the_test()
{
    LocaleTest::setLocale(LC_ALL, 'ru_RU.UTF-8', function () {
        // test content here
    });
}

```

Reply:
I followed your suggestion. But i did need to make some changes to make it work.
The name setLocale is already taken with other parameters.

Filename changed from LocaleTest to Locale

I am not sure if i did everythink correct. Please have a look.